### PR TITLE
tickets/DM 47783

### PR DESCRIPTION
**Pull request by couger01**
- **Use ESS framework to communicate with device; Publish sensor data so that ESS CSC can use it**
- **Split the SerialTemperatureScanner to use two servers providing ESS with data and the other to control the fan**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ __version__ = "{version}"
 asyncio_mode = "auto"
 
 [project.scripts]
-run_audiotrigger = "lsst.ts.audiotrigger:execute_runner"
+run_serial_temperature_scanner = "lsst.ts.audiotrigger:execute_serial_temperature_scanner"
```
Comment (wvreeven) [inline pyproject.toml:32]: Any reason why one is named `run_...` and the other `execute_...`? I'd suggest to name them both `run_...` so it is clear that they belong together.

File: python/lsst/ts/audiotrigger/constants.py [hunk 2]
```diff
@@ -19,11 +19,14 @@
...
 
 MAX_PI_GPIOS = 32
 """Maximum number of pi GPIO pins."""
 SLEEP = 1
 """Default sleep timer."""
 THRESHOLD = 10
 """Threshold over which misalignment of laser beam occurs."""
+MIN = 19
+MAX = 25
+SENSOR_INDEX = 0
```
Comment (wvreeven) [inline python/lsst/ts/audiotrigger/constants.py:32]: Please add doc strings for these variables. Also, `MIN` and `MAX` are very generic words. Please rename them to something meaningful.

File: python/lsst/ts/audiotrigger/read_serial_temp_scanner.py [hunk 3]
```diff
@@ -19,56 +19,77 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-__all__ = ["SerialTemperatureScanner"]
+__all__ = ["SerialTemperatureScanner", "execute_serial_temperature_scanner"]
```
Comment (wvreeven) [inline python/lsst/ts/audiotrigger/read_serial_temp_scanner.py:22]: See my comment for `pyproject.toml`.

File: python/lsst/ts/audiotrigger/read_serial_temp_scanner.py [hunk 4]
```diff
@@ -19,56 +19,77 @@
...
 
 
-class SerialTemperatureScanner(tcpip.OneClientServer):
+async def callback(data):
+    SerialTemperatureScanner.data = data
+
+
+def execute_serial_temperature_scanner():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--simulate", action="store_true")
```
Comment (wvreeven) [inline python/lsst/ts/audiotrigger/read_serial_temp_scanner.py:45]: Are you sure you want this argument to default to `True`? That means that this needs to be started with `--simulate False` which is what we normally do not do. Please set the default value to `False`.

File: python/lsst/ts/audiotrigger/read_serial_temp_scanner.py [hunk 5]
```diff
@@ -89,75 +110,69 @@ class SerialTemperatureScanner(tcpip.OneClientServer):
...
         else:
             self.pi = pigpio.pi()
+            self.serial = controller.device.RpiSerialHat(
+                name="Fan Control sensors",
+                device_id="/dev/ttyUSB0",
+                sensor=common.sensor.TemperatureSensor(log=self.log, num_channels=8),
+                baud_rate=19200,
+                callback_func=callback,
+                log=self.log,
+            )
```
Comment (wvreeven) [inline python/lsst/ts/audiotrigger/read_serial_temp_scanner.py:159]: Please rewrite this if/else so common parts (like the device_id) are only stored once. The only difference really is the sensor name. I think the mock sensor can handle the baud rate, can't it?

Comment (wvreeven) [reply python/lsst/ts/audiotrigger/read_serial_temp_scanner.py:159]: Please create a Jira ticket to send the config and add a TODO here. I requested that during our 1n1 on Monday.

**Review by wvreeven — CHANGES_REQUESTED** [review]
I think it looks good but there are a few things I'd like see improved before approving this PR. Please have a look at my comments and re-request my review when done.

File: pyproject.toml (hunk 1 above)
Comment (couger01) [reply pyproject.toml:32]: I think I just have that pattern by habit. But you're right about consistency so I will do it.

File: python/lsst/ts/audiotrigger/read_serial_temp_scanner.py (hunk 5 above)
Comment (couger01) [reply python/lsst/ts/audiotrigger/read_serial_temp_scanner.py:159]: https://github.com/lsst-ts/ts_audiotrigger/blob/8c091bc2e61feb725472fbf00de9026683f3abba/python/lsst/ts/audiotrigger/read_serial_temp_scanner.py#L175
I have the ticket and TODO here.

**Review by wvreeven — APPROVED** [review]
LGTM

**Pull request merged**